src/csv_reporter.cc

```diff
@@ -38,21 +39,51 @@ std::vector<std::string> elements = {
 
 bool CSVReporter::ReportContext(const Context& context) {
   PrintBasicContext(&GetErrorStream(), context);
+  return true;
+}
 
+void CSVReporter::ReportRuns(const std::vector<Run> & reports) {
```

Comment:
This still seems all wrong.  I think we need to make the CSV reporter store every reported run and only report them inside `Finalize`. That would mean the CSV reporter wouldn't be usable as a console reporter, instead it would be limited to outputting to files. @dominichamon Does this seem reasonable to you? If so I'm willing to fix this post-commit.